sympy/polys/rootoftools.py

```diff
@@ -772,13 +775,50 @@ def _get_roots(cls, method, poly, radicals):
         free_names = {str(i) for i in poly.free_symbols}
         for x in chain((symbols('x'),), numbered_symbols('x')):
             if x.name not in free_names:
-                poly = poly.xreplace({d: x})
+                poly = poly.replace(d, x)
                 break
+
+        if dom.is_QQ or dom.is_ZZ:
+            return cls._get_roots_qq(method, poly, radicals)
+        if dom.is_AlgebraicField:
+            return cls._get_roots_alg(method, poly, radicals)
+        else:
```

Comment:
Gaussian domains also need to be handled here:
```python
In [19]: all_roots(x**2 + I, extension=True)
---------------------------------------------------------------------------
NotImplementedError 
```

Reply:
Done!